### DISCUSSION TITLE
shadow-forks / fork-less PRs

### DISCUSSION THREAD

**Discussion by TitanNano** [community/community#45345] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

I have to say I'm quite fed up with having to create a fork of a repository every time I want to create a PR to a project. Forking a repository has its place and purpose, but I do not need one for every project I want to contribute to. It also clutters my Profile with repositories I don't actually want there.

It would be much better if GitHub added a sort of shadow forking functionality. I haven't fully fleshed this out, but the broad idea would be that I can clone the original repo and as soon as I push a branch to this remote, GitHub starts maintaining a private space for my branches on this repository. 

I won't have to fork the repository on my own, and it won't show up on my profile as a new repository. If I want to access my "fork", I can do so by going to the original repository. 
In case I really wish to create a fork of a repository at some point, GitHub can ask me if I would like to move my private branches to this fork. 
Should I at some point gain write access to the repository, GitHub can also automatically move my branches into the repository.

**Comment by joshka** [comment]
I want the same feature, but I'm fine with having a fork exist for it, just not a fork per repo. My use case is updating downstream users of a library. Say there are hundreds of repos and I can automate the updates pretty easily as:
```
git remote add repo-name https://github.com/repo-owner/repo-name
git fetch repo-name
git switch -c repo-name repo-name/main

# ...perform update

git push origin repo-name
gh pr create
```
Right now this fails (and is impossible to represent on the github UI
```
❯ gh pr create

Creating pull request for joshka:tokio-console into main in tokio-rs/console

? Title chore: migrate to ratatui and update crossterm
? Body <Received>
? What's next? Submit
pull request create failed: GraphQL: Head sha can't be blank, Base sha can't be blank, Head repository can't be blank, No commits between tokio-rs:main and joshka:tokio-console, Head ref must be a branch, not all refs are readable (createPullRequest)
```
